### PR TITLE
Add diagnose for Test Setup issues/file issues, issue #20

### PR DESCRIPTION
**Pull request by jusasiiv**

### CODE REVIEW

**Comment by blockonomics** [comment]
Have we already commited file permissions changes or that is not required ?

**Comment by jusasiiv** [comment]
> Have we already commited file permissions changes or that is not required ?

Not committed permission changes.

**Comment by blockonomics** [comment]
Please commit permission changes as well

File: modules/gateways/blockonomics.php [hunk 1]
```diff
@@ -84,11 +83,43 @@ function reqListener () {
 				newBtn.disabled = false;
 			}
 
+			function reqListener () {
+				responseObj = {};
+				switch (this.status) {
+					case 200:
+						responseObj = JSON.parse(this.responseText);
+						break;
+					case 401:
```
Comment (blockonomics) [inline modules/gateways/blockonomics.php]: Lets have only one default,  and one message. 
Unable to locate/execute *test_setup_url*. Contact blockonomics support for help

File: modules/gateways/blockonomics.php [hunk 2]
```diff
@@ -84,11 +83,43 @@ function reqListener () {
...
+																	permissions and that they are copied correctly"
+				showTestResult(responseObj);
+			}
+
 			newBtn.onclick = function() {
 				var testSetupUrl = "$system_url" + "testSetup.php";
 				var oReq = new XMLHttpRequest();
 				oReq.addEventListener("load", reqListener);
 				oReq.open("GET", testSetupUrl);
+				oReq.timeout = 10000; // 10 seconds
```
Comment (blockonomics) [inline modules/gateways/blockonomics.php]: Try to reproduce timeout
- If its doesn't reproduce. remove timeout
- If it reproduces, set same message as reqListener

**Pull request merged**